### ISSUE TITLE
ResourceWarning: files opened in readers.read_bed are never closed

### ISSUE THREAD

**Issue by Isy89** [pyranges/pyranges#377]
Hi!
I was reading bed files with the read_bed function and I got a ResourceWarning.
It seams that in pyranges/readers.py function read_bed line 127 or 129 the file is opened but never closed causing ResourceWarning.
In case many files are opened this may result in OSError: [Errno 24] Too Many Open Files.

This could be fixed in the following way:

original:

```python
    if f.endswith(".gz"):
        import gzip

        first_start = gzip.open(f).readline().split()[1]
    else:
        first_start = open(f).readline().split()[1]
```

updated:

```python
    if f.endswith(".gz"):
        import gzip
        with gzip.open(f) as of:
          first_start = of.readline().split()[1]
    else:
        with open(f) as of:
            first_start = of.readline().split()[1]
```

Thanks for the great package!

**Comment by marco-mariotti** [comment]
Thanks @Isy89 for the catch and the fix!

I invite you to introduce these changes yourself, following the developer guide https://pyranges.readthedocs.io/en/master/developer_guide.html

In this way, your contribution will be recorded in the history of the repo.

If you decline, we'll do this ourselves in a few days.

Best

**Comment by Isy89** [comment]
Hi @marco-mariotti! 

I followed the Developer guid and opened a pull request. 

When running the tests, I sow two tests failing:

1) FAILED tests/unit/out/test_out.py::test_write_bigwig - Exception: Using the coverage method requires that pyrle is installed.

is there any reason why the pyrle is not included in the dependencies ?

2) FAILED tests/unit/out/test_out.py::test_write_bigwig - AttributeError: 'DataFrame' object has no attribute 'chromosomes'. Did you mean: 'Chromosome'?

This is derived by a problem in pyrle. I opened an issue there (https://github.com/pyranges/pyrle/issues/9)
